canonical/lxd-ui
# Snapshot actions [WD-3032]

## Done

- Use icons for snapshot actions - edit, restore, delete
- Allow user to opt out restoring the instance state when restoring a stateful snapshot
- Add edit feature to update the name and expiry date of a snapshot

Fixes [WD-3032](https://warthogs.atlassian.net/browse/WD-3032)

## QA

1. Run the LXD-UI:
    - On the demo server via the link posted by @webteam-app below. This is only available for PRs created by collaborators of the repo. Ask @lorumic or @edlerd for access.
    - With a local copy of this branch, run as described in the [Readme](https://github.com/canonical/lxd-ui#setting-up-for-development).
2. Perform the following QA steps:
    - Open the instance detail page, snapshots tab
    - Check the look and feel of snapshot actions (with icon buttons instead of text buttons)
    - Restore a stateful snapshot - with or without the instance state
    - Edit a snapshot (rename it)
    - Edit a snapshot (update its expiry date)
    - Edit a snapshot (rename it and update its expiry date)

[WD-3032]: https://warthogs.atlassian.net/browse/WD-3032?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ
